darkcluster/src/main/java/com/linkedin/darkcluster/impl/ConstantQpsDarkClusterStrategy.java

```diff
@@ -0,0 +1,132 @@
+
+  @Override
+  public boolean handleRequest(RestRequest originalRequest, RestRequest darkRequest, RequestContext requestContext) {
+    float sendRate = getPerHostSendRate();
+    _rateLimiter.setRate(sendRate, 1000, 1);
+    return addRequestToQueue(originalRequest, darkRequest, requestContext);
+  }
+
+  /**
+   * We won't create this strategy if this config isn't valid for this strategy. For instance, we don't want to create
+   * the ConstantQpsDarkClusterStrategy if there's no darkClusterPerHostInboundTargetRate or if the
+   * darkClusterPerHostInboundTargetRate is zero, because we'd be doing pointless work on every getOrCreate.
+   * Instead if will go to the next strategy (or NoOpDarkClusterStrategy).
+   *
+   * This is a static method defined here because we don't want to instantiate a strategy to check this. It cannot be a
+   * method that is on the interface because static methods on an interface cannot be overridden by implementations.
+   * @param darkClusterConfig
+   * @return true if config is valid for this strategy
+   */
+  public static boolean isValidConfig(DarkClusterConfig darkClusterConfig) {
+    // TODO: Commented out for testing in ei. Still finalizing what config values we'll use. Uncomment when design is finalized.
+    return true;
+    // return darkClusterConfig.hasDispatcherOutboundTargetRate() &&
+    //    darkClusterConfig.getDispatcherOutboundTargetRate() > 0;
+  }
+
+  /**
+   * TODO: javadocs
+   * @return
+   */
+  private float getPerHostSendRate() {
+    try {
+      // Only support https for now. http support can be added later if truly needed, but would be non-ideal
+      // because potentially both dark and source would have to be configured.
+      int numDarkClusterInstances = _clusterInfoProvider.getHttpsClusterCount(_darkClusterName);
+      int numSourceClusterInstances = _clusterInfoProvider.getHttpsClusterCount(_originalClusterName);
+      if (numSourceClusterInstances != 0) {
+        return (float) (_darkClusterPerHostInboundTargetRate * numDarkClusterInstances) / numSourceClusterInstances;
+      }
+
+      return 0F;
+    } catch (ServiceUnavailableException e) {
+      _notifier.notify(() -> new RuntimeException(
+          "PEGA_0020 unable to compute strategy for source cluster: " + _originalClusterName + ", darkClusterName: " + _darkClusterName, e));
+      // safe thing is to return 0 so dark traffic isn't sent.
+      return 0F;
+    }
+  }
+
+  /**
+   * TODO: javadocs
+   * @param originalRequest
+   * @param darkRequest
+   * @param requestContext
+   * @return
+   */
+  private boolean addRequestToQueue(RestRequest originalRequest, RestRequest darkRequest, RequestContext requestContext)
+  {
+    _rateLimiter.submit(new Callback<None>() {
+      @Override
+      public void onError(Throwable e) {
+        //
+      }
+
+      @Override
+      public void onSuccess(None result) {
+        _baseDarkClusterDispatcher.sendRequest(originalRequest, darkRequest, requestContext, 1);
```

Comment:
Ditto for magic number here

Reply:
+1

Reply:
private static final int NUM_REQUESTS_TO_SEND_PER_RATE_LIMITER_CYCLE = 1;